reconcile/terraform_repo.py

```diff
@@ -290,6 +287,14 @@ def calculate_diff(
             if self.params.validate_git:
                 self.check_ref(d.repository, d.ref)
 
+        # validate that only one repo is being modified in each MR
+        total_modifications = len(diff.add) + len(diff.change) + len(diff.delete)
+
+        if total_modifications > 1:
+            raise Exception(
+                "Only one repository can be modified per merge request, please split your change out into multiple MRs"
+            )
+
```

Comment:
move to top of function (after `L261`)? Avoid unnecessary gitlab requests

Reply:
Because...it broke my tests. On second thought, this is kind of a naive approach to determining how many repos are being acted on so I'm going to take another crack at it

Reply:
Updated to make use of the `merge_results` function to determine how many repos are being updated because the data from the diff doesn't correspond 1:1 with how many repos are being updated. Also moved the check up top to fail early and avoid those extra requests. Tradeoff is that the user may see output like `update_repo repo_a` in their pr check output but the integration will fail so it will be pretty clear that there was an error.